client/src/components/Workflow/Editor/Forms/makeSection.js

```diff
@@ -0,0 +1,227 @@
+import _l from "utils/localization";
+
+/** Visit input nodes and enrich by name/value pairs from server data */
+function _visit(head, head_list, output, node) {
+    const postJobActions = node.postJobActions;
+    head_list = head_list || [];
+    head_list.push(head);
+    for (const i in head.inputs) {
+        const input = head.inputs[i];
+        const action = input.action;
+        if (action) {
+            input.name = `pja__${output.name}__${input.action}`;
+            if (input.pja_arg) {
+                input.name += `__${input.pja_arg}`;
+            }
+            if (input.payload) {
+                for (const p_id in input.payload) {
+                    input.payload[`${input.name}__${p_id}`] = input.payload[p_id];
+                    delete input.payload[p_id];
+                }
+            }
+            const d = postJobActions[input.action + output.name];
+            if (d) {
+                for (const j in head_list) {
+                    head_list[j].expanded = true;
+                }
+                if (input.pja_arg) {
+                    input.value = (d.action_arguments && d.action_arguments[input.pja_arg]) || input.value;
+                } else {
+                    input.value = "true";
+                }
+            }
+        }
+        if (input.inputs) {
+            _visit(input, head_list.slice(0), output, node);
+        }
+    }
+}
+
+function _makeRenameHelp(name_labels) {
+    let help_section = `This action will rename the output dataset. Click <a href="https://galaxyproject.org/learn/advanced-workflow/variables/">here</a> for more information. Valid input variables are:`;
```

Comment:
Could we make this a Vue component ?

Reply:
Thank you for looking into it and I agree with your comments, but I don't think it should be part of this PR. Same goes for the help text component. The goal of this PR is to rewire the overall structure of how we build the workflow editor forms without relying on the legacy tool-form base code. Once this new structure is tested and merged, it will allow us to tackle individual items separately.